cmd/bootkube/render.go

```diff
@@ -196,3 +217,28 @@ func altNamesFromURLs(urls []*url.URL) *tlsutil.AltNames {
 	}
 	return &an
 }
+
+func pickDNSServiceIP(serviceCIDR string) (string, error) {
+	ip, ipnet, err := net.ParseCIDR(serviceCIDR)
+	if err != nil {
+		return "", err
+	}
+	// pick the 10th IP as a conventional default
+	for i := 0; i < 10; i++ {
+		incIPv4(ip)
```

Comment:
While we've resolved the need for matching render and start flags, this is one property that must be correct btw bootkube and the target hosts. I've added a message to print out the DNS service IP that should be used.

```
bootkube render --asset-dir=assets --api-servers=https://node1.example.com:443 --api-server-alt-names=DNS=node1.example.com
Host kubelet MUST use --cluster-dns=10.3.0.10
```